public/app/plugins/datasource/tempo/resultTransformer.ts

```diff
@@ -623,6 +624,39 @@ function transformToTraceData(data: TraceSearchMetadata) {
   };
 }
 
+export function formatTraceQLMetrics(data: TraceqlMetricsResponse) {
+  const frames = data.series.map((series) => {
+    const timeValues: number[] = series.samples.map((sample) => parseInt(sample.timestampMs, 10));
```

Comment:
Should we be more defensive in how we map response data? Something we should discuss, but not strictly necessary to be done in this PR. We expect data.series / samples to exist but honestly I think we should be more careful and support the edge case that perhaps the data has not returned correctly in case of a malformed response etc that is not caught as an error/warning somewhere already up the response chain.